### PR TITLE
Add Select

### PR DESCRIPTION
**Pull request by andrei-sa**

### CODE REVIEW

File: Include/Rover/Select.hpp [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+#include <vector>
+#include "Rover/Autobox.hpp"
+#include "Rover/Evaluator.hpp"
+#include "Rover/Range.hpp"
+
+namespace Rover {
+namespace Details {
+  struct AnyIndex {
+    template<typename T>
+    operator const T() const { return T(); }
```
Comment (kranar) [inline Include/Rover/Select.hpp]: Don't implement this method, it's not necessary and this implementation assumes a default constructor.

File: Include/Rover/Select.hpp [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      /*!
+        \param container Set-like container defining value_type, begin, and end.
+      */
+      explicit Select(const Container& container);
+
+      //! Creates a Select with a user-defined selector.
+      /*!
+        \param container Set-like container defining value_type, begin, and end.
+        \param selector Generator evaluating to an index in the array.
+
```
Comment (kranar) [inline Include/Rover/Select.hpp]: Remove blank line.

File: Include/Rover/Select.hpp [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+                         and integer indexing.
+      */
+      explicit Select(Container container);
+
+      //! Creates a Select with a user-defined selector.
+      /*!
+        \param container Array-like container defining value_type, begin,
+                         and integer indexing.
+        \param selector Generator evaluating to an index in the array.
+
```
Comment (kranar) [inline Include/Rover/Select.hpp]: Remove blank line.

File: Include/Rover/Select.hpp [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      /*!
+        \param container Map-like container defining mapped_type, begin, and end.
+      */
+      explicit Select(const Container& container);
+
+      //! Creates a Select with a user-defined selector.
+      /*!
+        \param container Map-like container defining mapped_type, begin, and end.
+        \param selector Generator evaluating to an index in the array.
+
```
Comment (kranar) [inline Include/Rover/Select.hpp]: Remove blank line.

**Review by kranar — APPROVED** [review]

File: Include/Rover/Select.hpp (hunk 1 above)
Comment (andrei-sa) [reply Include/Rover/Select.hpp]: Yep I added it for debugging, forgot to remove

**Review by kranar — APPROVED** [review]

**Pull request merged**